config/application.rb

```diff
@@ -126,6 +125,8 @@ class Application < Rails::Application
         end
       end]
     end
+    config.action_cable.url =
+      "#{config.action_controller.relative_url_root}#{Settings.rails.session_store.args.path}/cable"
```

Comment:
I'm not sure `#{Settings.rails.session_store.args.path}` is necessary.

But also, I'm wondering if the relative url root is also necessary if you use `config.action_cable.mount_path` instead of `url`. See https://guides.rubyonrails.org/action_cable_overview.html#in-app.

Reply:
When using `config.action_cable.mount_path` instead of `url` I believe it relocates the action cable server, mounting it on top of localhost:3000/csc108. 
Ex: if we did `config.action_cable.mount_path = "/csc108/cable"`, the server is now located at localhost:3000/csc108/csc108/cable. The client however attempts to connect to "localhost:3000/csc108/cable". 
When we don't specify the `mount_path`, but instead specify the `url` we avoid relocating the server and can properly target it.
In order for the client to know where the web socket is located (know about the Settings.rails.session_store.args.path section of the url), I need to specify it; otherwise it gets left out of the url that the client tries connecting to.

Reply:
Perhaps you misunderstood what I was getting at. Try specifying `mount_path = 'cable'`. If that doesn't work push the change and I'll take a look.